### PR TITLE
feat: Create moderation for existing post edit

### PR DESCRIPTION
**Pull request by AmarTrebinjac**
Flow for creating a moderation entry for when user wants to edit their existing post

### CODE REVIEW

**Review by sshanzel — APPROVED** [review]
Looks amazing! Great work on the tests!

File: src/common/post.ts [hunk 1]
```diff
@@ -250,17 +250,32 @@ export type CreateSourcePostModeration = Omit<
...
 ) => {
-  const newPost = con.getRepository(SourcePostModeration).create({
+  if (args.postId) {
+    const post = await con
+      .getRepository(Post)
+      .findOneByOrFail({ id: args.postId });
+
+    const isUnauthorizedEdit =
+      (post.type === PostType.Freeform && args.createdById !== post.authorId) ||
+      (post.type !== PostType.Freeform && args.createdById !== post.scoutId);
```
Comment (capJavert) [inline src/common/post.ts:267]: Why do we need `post.type` check here?

File: src/common/post.ts [hunk 2]
```diff
@@ -250,17 +250,32 @@ export type CreateSourcePostModeration = Omit<
...
+      throw new ForbiddenError('Cannot edit posts created by other users');
+    }
+  }
+
+  const newModerationEntry = con.getRepository(SourcePostModeration).create({
     ...args,
     status: SourcePostModerationStatus.Pending,
   });
-  return await con.getRepository(SourcePostModeration).save(newPost);
+  return await con.getRepository(SourcePostModeration).save(newModerationEntry);
```
Comment (capJavert) [inline src/common/post.ts:278]: It will create a new row in SourcePostModeration right?

**Review by capJavert — APPROVED** [review]
Just few questions, but overall good

File: src/common/post.ts (hunk 1 above)
Comment (AmarTrebinjac) [reply src/common/post.ts:267]: @capJavert This was the reason I asked you about yesterday who is the post creator. I understood you such as that when the post is scraped or written on our site, the "creator" is the author id, and for share/link it is scout id. 

Did I understand that correctly?

File: src/common/post.ts (hunk 2 above)
Comment (AmarTrebinjac) [reply src/common/post.ts:278]: Correct!

File: src/common/post.ts (hunk 1 above)
Comment (sshanzel) [reply src/common/post.ts:267]: We should just check whether the user is the `authorId`. We only allow editing on `freeform` anyway. Also, the scout part is for community picks. Even `share` post can have the right `authorId`.

Comment (AmarTrebinjac) [reply src/common/post.ts:267]: Talked with Lee in DMs and he clarified that we do not use scoutId in squad, and both share and freeform use authorId for the "creator".

WIll therefor only check if createdById === authorId

**Pull request merged**